### DISCUSSION TITLE
Future plans for optilink-splitter?

### DISCUSSION THREAD

**Discussion by CM000n** [philippoo66/optolink-splitter#31] [Ideas]
Hi @philippoo66 😊 In the meantime I've been able to gain some experience with your great project here. Thanks again for your effort and great work!
I was wondering if you have any future plans for this project and if so, what are they?

**Background**:
I was wondering whether it might be useful to rewrite the optilink-splitter as a CLI? This would make the transfer of parameters more flexible and facilitate the packaging of optilink-splitter. Then you could also simply release it on PyPi and install it via pip or run it as a .pex (Python executable). This would possibly make the optilink-splitter more accessible to an even wider audience?
Of course, this would also entail more refactoring and a change of habits for current users. 
If you are interested, I could create a proposal and merge request. If you say you're happy with the current project structure and it's all too much overhead for you then that's ok too ☺️ Then I'll keep my feet still for now 😉

**Comment by philippoo66** [comment]
Good mornig @CM000n !

Sorry for my late reply, yesterday I spent at Repair Cafe... First of all thank you very much for your kind offer!

We just did this with the [open3e](https://github.com/open3e/open3e) and for sure for the more complex project this serves certain advantages. But we also stumbled into some traps, and even though I did a lot of the original code, I was not able to solve those issues. Also the need of using the _pip install --editable_ mode makes it more 'complicated' to me to do changes / enhancements (even I might get used to it...)

Plainly spoken if you would do it, you will be responsible for maintenance (of the pip install part) ;-)

As one of the big issues with 'making is simple' I see the datapoints address thing. It always requires to have a 'certain background' to get the thing running (as long as the target system is not a Vitodens 300 HO2B). For editing the settings_ini you need to have an understanding of the pip install mechanism to 'locate' the _right_ settings_ini and you need to edit it in sudo mode (as far as I remember).

So by making the process of installation easier, you pay with higher challenge of 'adjusting the settings'. 

What are the challenges today? There are two requirements - pyserial and phao-mqtt. On the recent Raspbian OSs you run into the "externally managed system" / venv thing. Both is described in the recent ReadMe/Wiki. With the pip installable you would instead need to describe the editing of the settings. Will it be possible to do the serial console / serial interface settings (raspi-config...) by pip install?

So if we (you ;-) would do it, I think we schould do it as an 'alternative option' and be aware of that this will require additional maintenance, testing and 'documentation' effort.

What do you think?

**Reply by CM000n** [reply]
> For editing the settings_ini you need to have an understanding of the pip install mechanism to 'locate' the right settings_ini and you need to edit it in sudo mode (as far as I remember)

My idea would have been to simply send the CLI a path to a .csv or .yaml file containing the data point addresses and lengths etc. This could then be located anywhere (on the host system).
The remaining parameters would simply be passed to the CLI as required. So there would no longer be a `settings_ini.py`. 😉 
I might actually make a merge request. That might make it easier to explain how I imagine it and would also make the discussion easier.

**Reply by philippoo66** [reply]
hi Simon @CM000n !

I realized you did a lot of work - thank you!! unfortunately curently I'm in a huge stress 
so I did not find the time to take a deeper look... I will do as soon as possible!

greetings & thanks again!
Phil

**Reply by CM000n** [reply]
No need to rush 😊 Take all the time you need.
In any case, thank you in advance for your feedback.
